### ISSUE TITLE
str function is not rewinding the body

### ISSUE THREAD

**Issue by gmponos** [guzzle/psr7#101]
If I use the function str(MessageInterface $message) I have to manually rewind the body of the message.

I am just wondering. Is this the desirable behaviour? I was expecting that it would automatically rewind the message body.

**Comment by Tobion** [comment]
I think the less magic the better. `Stream::__toString` does not rewind the body after returning the contents either. So I don't think `str` should rewind the body.

**Comment by gmponos** [comment]
OK.. I got you point. But what will happen at this line of code?
https://github.com/guzzle/guzzle/blob/master/src/MessageFormatter.php#L88

You see... I have created my own custom logger middleware and I copy pasted some things from there.
After my LoggerMiddelware was reading the message the next middleware had an empty response. Maybe it's something wrong I did on my middleware I haven't tested this MessageFormatter.
